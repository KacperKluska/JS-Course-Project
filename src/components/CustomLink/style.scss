.link {
  text-decoration: none;
  color: white;
  display: flex;
}
