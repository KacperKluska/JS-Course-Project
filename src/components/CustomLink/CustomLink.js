import { Link } from 'react-router-dom';
import './style.scss';

export default function CustomLink({ path, children }) {
  return (
    <Link className="link" to={path}>
      {children}
    </Link>
  );
}
