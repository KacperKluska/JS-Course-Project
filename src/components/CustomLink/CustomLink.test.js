import { BrowserRouter as Router, Link } from "react-router-dom";
import CustomLink from "./CustomLink";

/**
 * TODO
 * make more tests
 */
let wrapper;

beforeAll(() => {
  wrapper = mount(
    <Router>
      <CustomLink path="/testPage">Test me</CustomLink>
    </Router>
  );
});

test("renders", () => {
  expect(wrapper).not.toBe(null);
});

test("blocks tags", () => {
  expect(wrapper.find("a")).not.toBe(undefined);
  expect(wrapper.find(".link")).not.toBe(undefined);
});

test("values", () => {
  expect(wrapper.find("a").text()).toBe("Test me");
  expect(wrapper.find("a").props().href).toBe("/testPage");
});

test("CustomLink snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});
