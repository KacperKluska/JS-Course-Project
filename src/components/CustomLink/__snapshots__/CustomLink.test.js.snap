// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLink snapshot 1`] = `
<a
  className="link"
  href="/testPage"
  onClick={[Function]}
>
  Test me
</a>
`;
